src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs

```diff
@@ -0,0 +1,246 @@
+﻿using RevitServices.Persistence;
+using Autodesk.DesignScript.Runtime;
+using View = Revit.Elements.Views.View;
+using RevitServices.Transactions;
+using DynamoServices;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Revit.Elements.Views
+{
+    /// <summary>
+    ///     Base class for Revit Plan views
+    /// </summary>
+    [RegisterForTrace]
+    public class ScheduleView : View
+    {
+        #region Internal properties
+
+        /// <summary>
+        ///     An internal handle on the Revit element
+        /// </summary>
+        internal Autodesk.Revit.DB.ViewSchedule InternalViewSchedule
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        ///     Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalViewSchedule; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        ///     Private constructor
+        /// </summary>
+        private ScheduleView(Autodesk.Revit.DB.ViewSchedule view)
+        {
+            SafeInit(() => InitScheduleView(view));
+        }
+
+        /// <summary>
+        ///     Private constructor
+        /// </summary>
+        private ScheduleView(Category category, string name, ScheduleType type)
+        {
+            SafeInit(() => InitScheduleView(category, name, type));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        ///     Initialize a ScheduleView element
+        /// </summary>
+        private void InitScheduleView(Autodesk.Revit.DB.ViewSchedule view)
+        {
+            InternalSetScheduleView(view);
+        }
+
+        /// <summary>
+        ///     Initialize a ViewSchedule Element.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <param name="name"></param>
+        /// <param name="type"></param>
+        private void InitScheduleView(Category category, string name, ScheduleType type)
+        {
+            var doc = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(doc);
+
+            var vs = CreateViewSchedule(category, name, type);
+
+            InternalSetScheduleView(vs);
+
+            TransactionManager.Instance.TransactionTaskDone();
+
+            ElementBinder.CleanupAndSetElementForTrace(doc, this.InternalElement);
```

Comment:
What do you suggest? I just followed what you guys did with Section Views figuring that its how you want all the views to be handled. Feel free to post a sample of an alternative approach or a link to one.